llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -271,6 +272,27 @@ static void collectKernelModuleMap(
   }
 }
 
+// Go through function call graph searching for assert call.
+static bool hasAssertInFunctionCallGraph(llvm::Function *Func) {
+  std::vector<llvm::Function *> Workqueue;
+  Workqueue.push_back(Func);
```

Comment:
Will this lead to duplicated work when we have non-tree case like A->B A->C B->D C->D D->F?